data/src/main/java/com/antosik/benchproject/data/common/modules/MoviesModule.kt

```diff
@@ -44,6 +44,8 @@ object MoviesModule {
 
         @Provides
         fun provideMovieDatabase(app: Application): MovieDatabase =
-            Room.databaseBuilder(app, MovieDatabase::class.java, "movie_database").build()
+            Room.databaseBuilder(app, MovieDatabase::class.java, "movie_database")
+                .fallbackToDestructiveMigration()
```

Comment:
I used it because I had some problems like the below:
```
23:45:46.541  E  failed to add asset path '/data/app/~~MRjdnV0nFrcY0Y0ug77D0g==/com.antosik.benchproject-qQVZ4uKcya89Lm3ksvV8Rg==/base.apk'
                 java.io.IOException: Failed to load asset path /data/app/~~MRjdnV0nFrcY0Y0ug77D0g==/com.antosik.benchproject-qQVZ4uKcya89Lm3ksvV8Rg==/base.apk
                 	at android.content.res.ApkAssets.nativeLoad(Native Method)
                 	at android.content.res.ApkAssets.<init>(ApkAssets.java:296)
                 	at android.content.res.ApkAssets.loadFromPath(ApkAssets.java:145)
                 	at android.app.ResourcesManager.loadApkAssets(ResourcesManager.java:544)
                 	at android.app.ResourcesManager.loadApkAssets(ResourcesManager.java:513)
                 	at android.app.ResourcesManager.-$$Nest$mloadApkAssets(Unknown Source:0)
                 	at android.app.ResourcesManager$ApkAssetsSupplier.load(ResourcesManager.java:197)
                 	at android.app.ResourcesManager.createAssetManager(ResourcesManager.java:620)
                 	at android.app.ResourcesManager.createResourcesImpl(ResourcesManager.java:876)
                 	at android.app.ResourcesManager.findOrCreateResourcesImplForKeyLocked(ResourcesManager.java:946)
                 	at android.app.ResourcesManager.createResources(ResourcesManager.java:1411)
                 	at android.app.ResourcesManager.getResources(ResourcesManager.java:1602)
                 	at android.app.ResourcesManager.getResources(ResourcesManager.java:1519)
                 	at android.app.ResourcesManager.getResources(ResourcesManager.java:1507)
                 	at android.app.ActivityThread.getTopLevelResources(ActivityThread.java:2755)
                 	at android.app.ApplicationPackageManager.getResourcesForApplication(ApplicationPackageManager.java:2247)
                 	at android.app.ApplicationPackageManager.getResourcesForApplication(ApplicationPackageManager.java:2230)
                 	at android.app.ApplicationPackageManager.getDrawable(ApplicationPackageManager.java:1885)
                 	at android.app.ApplicationPackageManager.loadUnbadgedItemIcon(ApplicationPackageManager.java:3746)
                 	at android.app.ApplicationPackageManager.loadItemIcon(ApplicationPackageManager.java:3645)
                 	at android.content.pm.PackageItemInfo.loadIcon(PackageItemInfo.java:300)
                 	at android.content.pm.PackageItemInfo.loadIcon(PackageItemInfo.java:295)
                 	at android.app.ApplicationPackageManager.getApplicationIcon(ApplicationPackageManager.java:1956)
                 	at iwn.a(PG:5)
                 	at iwt.l(PG:1)
                 	at rc.agY(PG:61)
                 	at ivx.k(PG:3)
                 	at ivp.f(PG:3)
                 	at ivp.i(PG:3)
                 	at iwr.i(Unknown Source:0)
                 	at iwu.d(Unknown Source:10)
                 	at iwu.e(PG:3)
                 	at ive.apply(PG:161)
                 	at amxc.e(PG:2)
                 	at amxd.run(PG:9)
                 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
                 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
                 	at kkb.run(PG:5)
                 	at java.lang.Thread.run(Thread.java:1012)
```
And changing the database version or deleting the app didn't help.

Reply:
Some devices (mostly Samsung) are keeping app data even after deleting the app, in such cases, you need to go into app details and clear all data and cache. `fallbackToDestructiveMigration` is fine but only for development, for sure you should not release it to customers.

Reply:
Ok, I understand. I leave it, and when I would like to release this app (I think it is unlikely) I will delete `fallbackToDestructiveMigration`